src/frontend/src/btc/derived/has-pending-transactions.derived.ts

```diff
@@ -0,0 +1,66 @@
+import { pendingTransactionsStore } from '$btc/stores/btc-pending-transactions.store';
+import { LOCAL } from '$lib/constants/app.constants';
+import {
+	btcAddressMainnet,
+	btcAddressRegtest,
+	btcAddressTestnet
+} from '$lib/derived/address.derived';
+import { testnets } from '$lib/derived/testnets.derived';
+import { isNullish, nonNullish } from '@dfinity/utils';
+import { derived, type Readable } from 'svelte/store';
+
+export const initHasPendingTransactions = (address: string): Readable<boolean | 'loading'> =>
+	derived(
+		[btcAddressMainnet, btcAddressTestnet, btcAddressRegtest, testnets, pendingTransactionsStore],
+		([
+			$btcAddressMainnet,
+			$btcAddressTestnet,
+			$btcAddressRegtest,
+			$testnets,
+			$pendingTransactionsStore
+		]) => {
+			const pendingTransactionsData = $pendingTransactionsStore[address];
+
+			if (nonNullish(pendingTransactionsData)) {
+				return pendingTransactionsData.data.length > 0;
+			}
+
+			if (!$testnets) {
+				if (isNullish($btcAddressMainnet)) {
+					// Return loading while we don't have btc addresses and we can't tell
+					// whether there are pending transactions for a specific address.
+					return 'loading';
+				}
+				if ($btcAddressMainnet !== address) {
```

Comment:
Not sure if made the choice on purpose so just in case, you can inline/remove the above first check here:

```
if (nonNullish($btcAddressMainnet) && $btcAddressMainnet !== address) {
   return false;
}

return "loading";
```

Reply:
Done! Thanks!